### PR TITLE
[Runtimes] Remove enrichment for default gpu

### PR DESCRIPTION
**Pull request by Tankilevitch**
As part of enhancing function resources with defaults resources ( like cpu and memory ) we have also included the option for default gpu ( just as a backup if at some point in time gpu will be passed as default, which is super unlikely possibility ).

The GPU resource behaves differently from the CPU and Memory resources, the distinction is in the field name.
While CPU and Memory field names remain constant, the GPU field name might change depending on the GPU driver on the cluster.

This resulted in an issue on our end where we were referring to gpu as `nvidia.com/gpu`, which would cause a conflict if the user had a different gpu driver installed.

Because default GPU isn't currently available through Iguazio, and we're going to release a new MLRun version, I deleted the  enriching of the user's resources with default gpu.

### CODE REVIEW

File: mlrun/runtimes/utils.py [hunk 1]
```diff
@@ -507,6 +507,28 @@ def verify_requests(
...
+    resources = ["cpu", "memory"]
+    if not requirement:
+        return None, None
+
+    if len(requirement) > 3:
+        raise mlrun.errors.MLRunInvalidArgumentError(
+            "Unable to resolve the gpu type because there are more than 3 resources"
+        )
+    for resource, value in requirement.items():
+        if resource not in resources:
```
Comment (TomerShor) [inline mlrun/runtimes/utils.py:527]:
```suggestion
        if resource not in ["cpu", "memory"]:
```

File: mlrun/config.py [hunk 2]
```diff
@@ -545,13 +545,20 @@ def get_storage_auto_mount_params():
 
         return auto_mount_params
 
-    def get_default_function_pod_resources(self):
+    def get_default_function_pod_resources(
+        self, with_gpu_requests=True, with_gpu_limits=True
```
Comment (TomerShor) [inline mlrun/config.py:549]: Besides testing purposes, you only ever call this function with `(False, False)`.
Then why set the arguments' defaults to `True` and not `False`?

File: tests/test_config.py [hunk 3]
```diff
@@ -173,6 +173,84 @@ def test_decode_base64_config_and_load_to_object():
     assert decoded_list_output == expected_decoded_list_output
 
 
+def test_with_gpu_option_get_default_function_pod_resources(config):
```
Comment (TomerShor) [inline tests/test_config.py:176]: How about adding a test case with cpu/memory resources different than `None`?

**Review by TomerShor — CHANGES_REQUESTED** [review]
Nice one! Minor comments
